### PR TITLE
sysbuild: Minimise network core Kconfigs and image selection

### PR DESCRIPTION
**Pull request by nordicjm**
Makes selecting the network core image easier and allows for expansion by other modules by reducing multiple copies of the same code which was checking for hard-coded image names

### CODE REVIEW

File: sysbuild/CMakeLists.txt [hunk 1]
```diff
@@ -305,48 +305,11 @@ include(${CMAKE_CURRENT_LIST_DIR}/extensions.cmake)
...
-    elseif(SB_CONFIG_NETCORE_MULTIPROTOCOL_RPMSG)
-      add_overlay_config(
-        ${SB_CONFIG_NETCORE_MULTIPROTOCOL_RPMSG_NAME} # ToDo: Create a common Kconfig setting
-                                            # which can then default to the exact
-                                            # remote image selected, as to work for
-                                            # hci, mpsl, thread, etc.
-        "${secure_boot_image_dir}/secure_boot.conf"
-      )
-    endif()
+  if(SB_CONFIG_SECURE_BOOT_NETCORE AND NOT "${SB_CONFIG_NETCORE_NETCORE_NONE}")
```
Comment (57300) [inline sysbuild/CMakeLists.txt:308]:
```suggestion
  if(SB_CONFIG_SECURE_BOOT_NETCORE AND NOT SB_CONFIG_NETCORE_NONE)
```

File: sysbuild/Kconfig.netcore [hunk 2]
```diff
@@ -86,129 +86,40 @@ config NETCORE_MULTIPROTOCOL_RPMSG
...
-	  If not set, the board given to sysbuild is used.
-
-config NETCORE_802154_RPMSG_DOMAIN
-	string
-	default "CPUNET" if (NETCORE_802154_RPMSG_BOARD = "nrf5340dk_nrf5340_cpunet" || NETCORE_802154_RPMSG_BOARD = "thingy53_nrf5340_cpunet" || NETCORE_802154_RPMSG_BOARD = "nrf7002dk_nrf5340_cpunet")
-
-config NETCORE_802154_RPMSG_DOMAIN_APP
-	bool
-	default y if (NETCORE_802154_RPMSG_BOARD = "nrf5340dk_nrf5340_cpunet" || NETCORE_802154_RPMSG_BOARD = "thingy53_nrf5340_cpunet" || NETCORE_802154_RPMSG_BOARD = "nrf7002dk_nrf5340_cpunet")
+	  Image name of empty image.
```
Comment (57300) [inline sysbuild/Kconfig.netcore:99]:
```suggestion
	  Name of the netcore image.
```

File: sysbuild/Kconfig.netcore [hunk 3]
```diff
@@ -86,129 +86,40 @@ config NETCORE_MULTIPROTOCOL_RPMSG
...
-	default "multiprotocol_rpmsg"
-	depends on NETCORE_MULTIPROTOCOL_RPMSG
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/empty_net_core" if NETCORE_EMPTY
+	default "${ZEPHYR_BASE}/samples/bluetooth/hci_rpmsg" if NETCORE_HCI_RPMSG
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/bluetooth/rpc_host" if NETCORE_RPC_HOST
+	default "${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg" if NETCORE_802154_RPMSG
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/multiprotocol_rpmsg" if NETCORE_MULTIPROTOCOL_RPMSG
 	help
-	  Image name of radio hci rpmsg.
+	  Image name of empty image.
```
Comment (57300) [inline sysbuild/Kconfig.netcore:109]:
```suggestion
	  Source directory of the netcore image.
```

**Review by 57300 — APPROVED** [review]
Nice. Just a few comments, feel free to address them here or in the upmerge.

File: sysbuild/Kconfig.netcore (hunk 2 above)
Comment (nordicjm) [reply sysbuild/Kconfig.netcore]: Fixed

**Review by tejlmand — APPROVED** [review]

**Pull request merged**